messenger/make.py

```diff
@@ -67,9 +76,55 @@ def build_octave():
     do_build(make_cmd, 'messenger.mex')
 
 
+def which_matlab():
```

Comment:
Right.. the function which_matlab which I wrote doesn't reimplement which
at all. It calls "which matlab." However it usually isn't in the path, so
when that fails it looks in the default location on all 3 platforms.

I was actually inspired by your function to make it work on all 3
platforms. However, your code assumes the same structure of MATLAB on all 3
platforms, which for some reason isn't the case (why MATHWORKS?), which is
why I had to use if statements and handle each OS separately.

On Fri, Apr 10, 2015 at 4:41 PM, Sang Han <notifications@github.com> wrote:

> In messenger/make.py
> <https://github.com/arokem/python-matlab-bridge/pull/178#discussion_r28188377>
> :
>
> > @@ -67,9 +76,55 @@ def build_octave():
> >      do_build(make_cmd, 'messenger.mex')
> >
> >
> > +def which_matlab():
>
> pexept assumes the binaries are in the path since it replicates BSD utils
> which.
>
> However, get_matlab_bin should return absolute path of the matlab binary
> regardless of if it is in the path or not.
>
>
> https://github.com/jjangsangy/python-matlab-bridge/blob/master/messenger/make.py#L206
>
> Then running get_matlab_env with the path to the binary will also return
> a dictionary containing that particular configuration environment variables.
>
>
> https://github.com/jjangsangy/python-matlab-bridge/blob/master/messenger/settings.py#L15
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/arokem/python-matlab-bridge/pull/178/files#r28188377>.
>